### ISSUE TITLE
Move layout information from nodes constructor' to their Glow types

### ISSUE THREAD

**Issue by shajrawi** [pytorch/glow#3790]
with a default of 'any layout' for any node that's agonistic to layout constraints. straightforward / just some manual labor for someone motivated: During an operator's construction it will propagate the layout string to the type field instead of saving it as a string in the operator's class.
The type itself should, preferably, not contain the layout string but a pointer to a global "cache" of strings: this reduces the memory usage of types and makes type-hashing easier.

**Comment by mciprian13** [comment]
@shajrawi I agree with this approach of having the layout part of the Type definition. Is this work planned?

**Comment by shajrawi** [comment]
The issue is up for grabs / not assigned, please feel free to work on it if you're interested.

**Comment by gudipudiramanakumar** [comment]
@shajrawi can I work on this issue, I'm quite new to working on open source but I have some experience and skill in C/C++ programming.
